src/bin/bat/main.rs

```diff
@@ -56,15 +59,43 @@ fn run_cache_subcommand(matches: &clap::ArgMatches) -> Result<()> {
     Ok(())
 }
 
+fn get_syntax_mapping_to_paths(
+    mappings: Vec<(GlobMatcher, MappingTarget)>,
+) -> HashMap<String, Vec<String>> {
+    let mut map = HashMap::new();
+    for mapping in mappings {
+        match mapping.1 {
+            MappingTarget::MapToUnknown => {}
+            MappingTarget::MapTo(s) => {
+                let globs = map.entry(s.into()).or_insert(Vec::new());
+                globs.push(mapping.0.glob().glob().into());
+            }
+        }
+    }
+    map
+}
+
 pub fn list_languages(config: &Config) -> Result<()> {
     let assets = assets_from_cache_or_binary()?;
     let mut languages = assets
         .syntaxes()
         .iter()
         .filter(|syntax| !syntax.hidden && !syntax.file_extensions.is_empty())
+        .cloned()
         .collect::<Vec<_>>();
     languages.sort_by_key(|lang| lang.name.to_uppercase());
 
+    let configured_languages =
+        get_syntax_mapping_to_paths(config.syntax_mapping.mappings().clone());
+
+    for lang in languages.iter_mut() {
+        if configured_languages.contains_key(&lang.name) {
+            let additional_paths = configured_languages.get(&lang.name).unwrap();
```

Comment:
Instead of using `.contains_key` and then using `.get`, you can simply call `get` and match on the result - avoiding the `.unwrap()` call.